### PR TITLE
zig: enable cross-compile support

### LINKED ISSUE #24959: zig: cross-compile the toolchain
**Old info below**, I've made it possible to cross-compile without touching the `llvm10` template or nasty hacks. See the latest template [here](https://github.com/jjsullivan5196/void-packages/tree/zig-cross/srcpkgs/zig)

I've been trying to get the zig compiler cross-compiled to aarch64 (want to have it on the pinephone) but I'm getting snubbed by some linker errors. 

I took a look at the zig project's own [bootstrap package](https://github.com/ziglang/zig-bootstrap) and found that it needs an AVR codegen target for cross targets, and that the host zig compiler can be used to cross-compiler the later stages. I made appropriate changes to the [`llvm10` and `zig` templates](https://github.com/jjsullivan5196/void-packages/tree/zig-cross). On `aarch64` and `aarch64-musl` I can fully compile stage0 with `cross-gcc`, but I get this mess when it links

```
[ 97%] Linking CXX static library zig_cpp/libzigcompiler.a
[ 97%] Built target zigcompiler
Scanning dependencies of target zig0
[ 98%] Building CXX object CMakeFiles/zig0.dir/src/stage2.cpp.o
[ 98%] Building CXX object CMakeFiles/zig0.dir/src/main.cpp.o
[ 99%] Linking CXX executable zig0
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `stripComments(_xmlNode*) [clone .isra.0]':
(.text._ZL13stripCommentsP8_xmlNode.isra.0+0x54): undefined reference to `xmlUnlinkNode'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL13stripCommentsP8_xmlNode.isra.0+0x60): undefined reference to `xmlFreeNode'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `searchOrDefine(unsigned char const*, _xmlNode*)':
(.text._ZL14searchOrDefinePKhP8_xmlNode+0x15c): undefined reference to `xmlNewNs'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `llvm::windows_manifest::WindowsManifestMerger::WindowsManifestMergerImpl::~WindowsManifestMergerImpl()':
(.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImplD2Ev+0x24): undefined reference to `xmlFreeDoc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImplD2Ev+0x38): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImplD2Ev+0x3c): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `treeMerge(_xmlNode*, _xmlNode*)':
(.text._ZL9treeMergeP8_xmlNodeS0_+0xc4): undefined reference to `xmlNewProp'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x120): undefined reference to `xmlStrdup'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x180): undefined reference to `xmlCopyNamespace'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x52c): undefined reference to `xmlUnlinkNode'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x538): undefined reference to `xmlAddChild'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x74c): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x754): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x924): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZL9treeMergeP8_xmlNodeS0_+0x928): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `checkAndStripPrefixes(_xmlNode*, std::vector<_xmlNs*, std::allocator<_xmlNs*> >&)':
(.text._ZL21checkAndStripPrefixesP8_xmlNodeRSt6vectorIP6_xmlNsSaIS3_EE+0x2a8): undefined reference to `xmlFreeNs'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `llvm::windows_manifest::WindowsManifestMerger::WindowsManifestMergerImpl::getMergedManifest()':
(.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0x44): undefined reference to `xmlDocGetRootElement'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0x64): undefined reference to `xmlNewDoc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0x70): undefined reference to `xmlDocSetRootElement'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0x78): undefined reference to `xmlKeepBlanksDefault'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0x98): undefined reference to `xmlDocDumpFormatMemoryEnc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0xac): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0xb0): undefined reference to `xmlFree'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl17getMergedManifestEv+0xc4): undefined reference to `xmlFreeDoc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMWindowsManifest.a(WindowsManifestMerger.cpp.o): in function `llvm::windows_manifest::WindowsManifestMerger::WindowsManifestMergerImpl::merge(llvm::MemoryBuffer const&)':
(.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0x54): undefined reference to `xmlSetGenericErrorFunc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0x70): undefined reference to `xmlReadMemory'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0x84): undefined reference to `xmlSetGenericErrorFunc'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0xb0): undefined reference to `xmlDocGetRootElement'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0xe0): undefined reference to `xmlUnlinkNode'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0xec): undefined reference to `xmlFreeNode'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm16windows_manifest21WindowsManifestMerger25WindowsManifestMergerImpl5mergeERKNS_12MemoryBufferE+0x104): undefined reference to `xmlDocGetRootElement'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(Compression.cpp.o): in function `llvm::zlib::compress(llvm::StringRef, llvm::SmallVectorImpl<char>&, int)':
(.text._ZN4llvm4zlib8compressENS_9StringRefERNS_15SmallVectorImplIcEEi+0x40): undefined reference to `compressBound'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm4zlib8compressENS_9StringRefERNS_15SmallVectorImplIcEEi+0x6c): undefined reference to `compress2'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(Compression.cpp.o): in function `llvm::zlib::uncompress(llvm::StringRef, char*, unsigned long&)':
(.text._ZN4llvm4zlib10uncompressENS_9StringRefEPcRm+0x3c): undefined reference to `uncompress'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(Compression.cpp.o): in function `llvm::zlib::crc32(llvm::StringRef)':
(.text._ZN4llvm4zlib5crc32ENS_9StringRefE+0x14): undefined reference to `crc32'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(CRC.cpp.o): in function `llvm::crc32(unsigned int, llvm::ArrayRef<unsigned char>)':
(.text._ZN4llvm5crc32EjNS_8ArrayRefIhEE+0x38): undefined reference to `crc32'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(DynamicLibrary.cpp.o): in function `llvm::sys::DynamicLibrary::HandleSet::~HandleSet()':
(.text._ZN4llvm3sys14DynamicLibrary9HandleSetD2Ev+0x24): undefined reference to `dlclose'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm3sys14DynamicLibrary9HandleSetD2Ev+0x38): undefined reference to `dlclose'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(DynamicLibrary.cpp.o): in function `llvm::sys::DynamicLibrary::HandleSet::DLOpen(char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
(.text._ZN4llvm3sys14DynamicLibrary9HandleSet6DLOpenEPKcPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x14): undefined reference to `dlopen'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm3sys14DynamicLibrary9HandleSet6DLOpenEPKcPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x2c): undefined reference to `dlerror'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(DynamicLibrary.cpp.o): in function `llvm::sys::DynamicLibrary::HandleSet::DLClose(void*)':
(.text._ZN4llvm3sys14DynamicLibrary9HandleSet7DLCloseEPv+0x0): undefined reference to `dlclose'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(DynamicLibrary.cpp.o): in function `llvm::sys::DynamicLibrary::HandleSet::DLSym(void*, char const*)':
(.text._ZN4llvm3sys14DynamicLibrary9HandleSet5DLSymEPvPKc+0x0): undefined reference to `dlsym'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: /usr/aarch64-linux-gnu/usr/lib/libLLVMSupport.a(Process.cpp.o): in function `llvm::sys::Process::FileDescriptorHasColors(int)':
(.text._ZN4llvm3sys7Process23FileDescriptorHasColorsEi+0xa0): undefined reference to `setupterm'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm3sys7Process23FileDescriptorHasColorsEi+0xcc): undefined reference to `tigetnum'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm3sys7Process23FileDescriptorHasColorsEi+0xd8): undefined reference to `set_curterm'
/usr/lib/gcc/aarch64-linux-gnu/9.3.0/../../../../aarch64-linux-gnu/bin/ld: (.text._ZN4llvm3sys7Process23FileDescriptorHasColorsEi+0xdc): undefined reference to `del_curterm'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/zig0.dir/build.make:309: zig0] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/zig0.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 99%] Built target zig_build_libstage2
make: *** [Makefile:149: all] Error 2
=> ERROR: zig-0.6.0_1: do_build: '${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at common/build-style/cmake.sh:74
```

This version of `llvm10` can still build zig on the host target (x86_64) without a problem. Not sure what's causing this to happen. Has anyone seen errors like this trying to build other llvm10-based software, maybe rust? @Duncaen @ifreund @q66

### PR DESCRIPTION
**Pull request by jjsullivan5196**
Closes #24959

### CODE REVIEW

File: srcpkgs/zig/template [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Template file for 'zig'
 pkgname=zig
 version=0.6.0
-revision=1
-archs="x86_64*"
+revision=2
+archs="x86_64* aarch64*"
```
Comment (ericonr) [inline srcpkgs/zig/template:5]: Is it still that limited? :/

Also, if the final result for `x86_64` doesn't change, you don't need to revbump, since changes to the template will trigger the build for aarch64 anyway.

Comment (jjsullivan5196) [reply srcpkgs/zig/template:5]: @ericonr I only tested aarch64, but I guess travis will test most stuff. Do you think I should change `archs` and see what happens?

Also, I'll decrease the revbump along with it.

Comment (ericonr) [reply srcpkgs/zig/template:5]: I think it'd make sense. Ideally also test programs built using it.

Comment (jjsullivan5196) [reply srcpkgs/zig/template]: Did as much as I could about this, I'll give more details in further comments.

**Comment by jjsullivan5196** [comment]
I tested cross-compiling for void's supported targets. All of x64, i686, and arm/aarch64 are working. However, I couldn't get cross-builds working for mips or ppc, and I don't have the hardware to see if they'll compile natively. I'm not sure where the problem is at this point, but I recorded the build logs for all of them and have them [attached](https://github.com/void-linux/void-packages/files/5254628/build-logs.tar.gz).

@ericonr All ready for you if you still want to try armv6

**Comment by ericonr** [comment]
Tagging interested parties on the platform side: @pullmoll and @q66 

And thanks, I will try to build it!

**Comment by ericonr** [comment]
I get an illegal instruction error, code 132. Just running `zig` is enough to trigger it.

**Comment by ericonr** [comment]
The build is populating `/tmp` with artifacts. `zig` should probably be pointed elsewhere for cache.

File: srcpkgs/zig/template [hunk 2]
```diff
@@ -11,9 +11,57 @@ license="MIT"
...
+	# TODO _mcpu might be a candidate for cross-profiles
+	#
+	# Set _cross_triple and _mcpu for target triples that are not directly
+	# available in zig's targets.
+	#
+	# List targets with: zig targets
+
+	case "${XBPS_CROSS_TRIPLET}" in
+		armv7l*)
+			_cross_triple="arm-linux-${XBPS_CROSS_TRIPLET#arm*-linux-*}"
```
Comment (ericonr) [inline srcpkgs/zig/template:27]: Why is this necessary?

**Comment by ericonr** [comment]
Partial diff, the `v6k` part is potentially sketchy.

```diff
diff --git a/srcpkgs/zig/template b/srcpkgs/zig/template
index 157b787f6f..10ea3be846 100644
--- a/srcpkgs/zig/template
+++ b/srcpkgs/zig/template
@@ -22,13 +22,17 @@ if [ "${CROSS_BUILD}" ]; then
 	#
 	# List targets with: zig targets
 
-	case "${XBPS_CROSS_TRIPLET}" in
+	case "${XBPS_TARGET_MACHINE}" in
+		armv6l*)
+			_cross_triple="${XBPS_CROSS_TRIPLET}"
+			_mcpu="generic+v6k"
+			;;
 		armv7l*)
-			_cross_triple="arm-linux-${XBPS_CROSS_TRIPLET#arm*-linux-*}"
+			_cross_triple="${XBPS_CROSS_TRIPLET/armv7l/arm}"
 			_mcpu="generic+v7a"
 			;;
 		i686*)
-			_cross_triple="i386-linux-${XBPS_CROSS_TRIPLET#i686*-linux-*}"
+			_cross_triple="${XBPS_CROSS_TRIPLET/i686/i386}"
 			_mcpu="baseline"
 			;;
 		*)
```

**Comment by ericonr** [comment]
Hmm

```
[ericonr@voidzero test]$ zig build-exe hello.zig
Semantic Analysis [667/867] ./hello.zig:4:38: error: no member named 'writer' in struct 'std.fs.file.File'
    const stdout = std.io.getStdOut().writer();
                                     ^
/usr/lib/zig/std/start.zig:253:40: note: referenced here
            const result = root.main() catch |err| {
                                       ^
```

**Comment by jjsullivan5196** [comment]
@ericonr would this work?

```zig
const debug = @import("std").debug;

pub fn main() void {
  debug.warn("Hello, world\n", .{});
}
```

File: srcpkgs/zig/template (hunk 2 above)
Comment (jjsullivan5196) [reply srcpkgs/zig/template:27]: Some of zigs targets don't exactly match up with gcc triplets. In this case, all of arm32 is under `arm-...` with the different subarchs identified via the mcpu parameter.

**Comment by ericonr** [comment]
That does work.

File: srcpkgs/zig/template (hunk 2 above)
Comment (ericonr) [reply srcpkgs/zig/template:27]: Fair enough. I think the substitution I'm doing below is a bit neater, if you'd like to use that.

**Comment by jjsullivan5196** [comment]
> The build is populating `/tmp` with artifacts. `zig` should probably be pointed elsewhere for cache. (either inside builddir or a cache in hostdir, if it makes sense)

I've seen ccache options in the cmake files, I'll take a look at this.

**Comment by jjsullivan5196** [comment]
I got the latest findings reflected in the template, the enabled archs should build+run properly, and zig won't dump its cache in `/tmp` anymore. I also restricted arm32 to v6 and up, since `_mcpu="generic+v5te"` emitted this error:

```
<inline asm>:1:2: error: instruction requires: armv6k
        yield
        ^
LLVM ERROR: Error parsing inline asm
```

File: srcpkgs/zig/template (hunk 2 above)
Comment (jjsullivan5196) [reply srcpkgs/zig/template]: cleaned this part up

**Comment by ericonr** [comment]
That's the error I got with `generic+v6` too.

**Comment by pullmoll** [comment]
In `qt5-webengine` we replace `yield` with `nop` and this seems to be sufficient. Of course `yield` would force a task switch for SMP, while a `nop` does not. But then I think there are no `armv5tel` multi-cores and few, if any, `armv6l`?

**Comment by jjsullivan5196** [comment]
@pullmoll may be simple enough to write a patch to do it then, I'll take a look soon

**Comment by ericonr** [comment]
@jjsullivan5196 probably best to just do a sed substitution, like the `qt5-webengine` template does c:

**Comment by Piraty** [comment]
upstream refuses to support cross-building their "stage1 zig" (see: https://github.com/ziglang/zig/pull/4494#issuecomment-587172905).

I propose to add the complete list of supported architectures to`archs=` and let everyone in need of zig on a non x86 platform build it manualy (I guess < 5 users).

**Comment by q66** [comment]
if the cross-compiled artifacts here are not stage1, we definitely should not enable it

**Comment by jjsullivan5196** [comment]
@q66 @Piraty gcc always compiles stage1 in this template, stage1 compiles stage2 for a native build, or stage2 on host compiles stage2 for target. See here for more details https://github.com/ziglang/zig-bootstrap

**Comment by jjsullivan5196** [comment]
To address some confusion: all artifacts of this template are the stage2 compiler, cross-compiled or not. stage1 is never part of the final output.

**Comment by Piraty** [comment]
@ifreund said on irc (not sure how accurate that is though)
> ifreund: stage2 didn't ship with 0.6.0, real work on it started just after that release
> ifreund: 0.7.0 will ship with an incompleted stage2, the goal is to have a complete stage2 for 0.8.0
> ifreund: 0.7.0 should be released in the month after llvm 11 releases fyi

According to [zig-0.6.0 README](https://github.com/ziglang/zig/blob/0.6.0/README.md):

> Note: Stage 2 compiler is not complete. Beta users of Zig should use the Stage 1 compiler for now.

The way i see it, the current template builds stage1 by just calling `cmake ; make`, stage2 is not part of this template yet.
In the future, this template should be renamed to `zig-bootstrap` or similar and be used by the real stage2 zig (which could then maybe have it's own `build_style=zig`)

**Comment by jjsullivan5196** [comment]
@Piraty my bad, I think I was getting the terms mixed up. I edited my previous comments to be accurate

**Comment by ericonr** [comment]
Ping

**Comment by jjsullivan5196** [comment]
@ericonr sorry 😅 I dropped the ball on this one due to other interests. However, I did get a patch merged upstream that should make this easier. At the least, I'll get this rebased so it works with the latest stable zig

**Comment by ericonr** [comment]
@jjsullivan5196 ping?

**Comment by jjsullivan5196** [comment]
@ericonr definitely dead! sorry, should have closed earlier. If anyone wants to take this the rest of the way feel free. [This should make things a little easier, not sure where the upstream is at this point.](https://github.com/ziglang/zig/pull/6387)
